File: Algo/hr_implement/3/permutationEquation.js
function permutationEquation(p) {
    let result = [];
    for(let i=0;i<p.length;i++) {
        let x = i + 1;
        let index = p.indexOf(x);
        x = p.indexOf(index + 1);
        result.push(x+1)
    }
    return result;
}

let p = [2,3,1];
console.log(permutationEquation(p));

File: Algo/hr_implement/3/permutationEquation.py
"""
For each x from 1 to n, print an integer denoting any valid y satisfying the equation p(p(y)) = x on a new line.
"""

def permutationEquation(p):
	result = []
	for i in range(len(p)):
		x = i + 1
		first_index = p.index(x)
		x = p.index(first_index + 1)
		result.append(x+1)	
	return result

if __name__ == "__main__":
	p = [2,3,1]
	p = [5,2,1,3,4]
	print permutationEquation(p)